actioncable/lib/action_cable/subscription_adapter/redis.rb

```diff
@@ -0,0 +1,45 @@
+begin
+  require 'em-hiredis'
+  require 'redis'
+rescue Gem::LoadError => e
+  raise Gem::LoadError, "You are trying to use the Redis Action Cable adapter, but do not have the proper gems installed. Add `gem 'em-hiredis'` and `gem 'redis'` to your Gemfile (and ensure its version is at the minimum required by Action Cable)."
+end
```

Comment:
Would consider moving this into a class macro in the adapters themselves, so we can share the raise and message behavior.

How about:

```ruby
module ActionCable
  module SubscriptionAdapter
    class Base
      def self.require_gems!(*gems)
        gems.each { |dependency| require dependency }
      rescue Gem::LoadError => error
        raise error, "You're trying to use the #{name} Action Cable pubsub adapter, but you're missing " \
          "#{gems.join(', ')} from your Gemfile."
      end
    end
    
    class Postgres < Base
      require_gems! 'pg'
    end
    
    class Redis < Base
      require_gems! 'em-hiredis', 'redis'
    end
  end
end
```